ccmlib/containers/docker_client.py

```diff
@@ -0,0 +1,143 @@
+            self.container.remove(force=True)
+        except docker.errors.APIError as err:
+            LOGGER.warning("Failed to remove '{}' container".format(self.image), exc_info=err)
+        self.container = self.container_name = self.container_pid = None
+        LOGGER.info("The'{}' container is closed".format(self.name))
+
+    def execute_command(self, command, is_stream=True, **kwargs):
+        if self.container is None:
+            raise docker.errors.DockerException("container is not running".format(self.name, command))
+
+        command = f"{self.shell_type} -c \"{command}\""
+        LOGGER.debug("Execute '{}' command in '{}' container".format(command, self.name))
+        try:
+            output = self.container.exec_run(command, stream=is_stream, **kwargs)[1]
+        except docker.errors.APIError as err:
+            if not self.is_running():
+                self.close()
+            raise docker.errors.DockerException("Docker container API error: {}".format(err))
+        if is_stream:
+            response = "".join([line.decode("utf-8", "ignore").rstrip() for line in output])
+            LOGGER.debug("The response of '{}' command is:\n{}".format(command, response))
+            output = response
+
+        return output
+
+    def is_running(self):
+        """
+        Check if class container state is running
+        :return: boolean
+        """
+        try:
+            return self.container.status == "running"
+        except docker.errors.APIError:
+            return False
+
+    @staticmethod
+    def server_ip(destination_address="8.8.8.8"):
+        socket_connection = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        try:
+            for _ in range(5):
+                socket_connection.connect((destination_address, randrange(50000, 60000)))
+                break
+        except OSError:
+            pass
+        finally:
+            ip = socket_connection.getsockname()[0]
+        return ip
+
+    def stop_service(self, service_name):
+        """
+        This method stops the given service name
+        :param service_name: str name of the service
+        """
+        LOGGER.info(f"Stopping the service {service_name} in container {self.name}")
+        self.execute_command(f"service {service_name} stop")
+
+    def start_service(self, service_name):
+        """
+        This method starts the given service name
+        :param service_name: str name of the service
+        """
+        LOGGER.info(f"Starting the service {service_name} in container {self.name}")
+        self.execute_command(f"service {service_name} start")
+
+    def restart_service(self, service_name):
+        """
+        This method restarts the given service name
+        :param service_name: str name of the service
+        """
+        LOGGER.info(f"Restarting the service {service_name} in container {self.name}")
+        self.execute_command(f"service {service_name} restart")
+
+    @property
+    def logs(self):
```

Comment:
I think this would be good enough, since we have the `mark_log` functionally which is very important for dtests, and getting the logs like that won't play nice with it.

we need a file that keep getting written to

Reply:
I've handled logging in #262 already (borrowed it from SCT)